--- a/configure
+++ b/configure
@@ -18789,7 +18789,7 @@
 				*) ;;	# Assume all other versions are good.
 				esac
 				;;
-			30*)
+			30*|40*)
 				# OpenSSL 3; we use the 1.1x API
 				# https://openssl.org/policies/general/versioning-policy.html
 				CPPFLAGS="$CPPFLAGS -DOPENSSL_API_COMPAT=0x10100000L"
--- a/configure.ac
+++ b/configure.ac
@@ -2891,7 +2891,7 @@
 				*) ;;	# Assume all other versions are good.
 				esac
 				;;
-			30*)
+			30*|40*)
 				# OpenSSL 3; we use the 1.1x API
 				# https://openssl.org/policies/general/versioning-policy.html
 				CPPFLAGS="$CPPFLAGS -DOPENSSL_API_COMPAT=0x10100000L"
