--- /var/folders/q6/c869d13s0v36hrh8b5_6zq9m0000gn/T/opencode/openssh-pkcs11/ssh-pkcs11.c	2024-09-20 01:20:48
+++ /var/folders/q6/c869d13s0v36hrh8b5_6zq9m0000gn/T/opencode/openssh-pkcs11/ssh-pkcs11.c.new	2026-06-09 20:32:21
@@ -786,8 +786,8 @@
 		ossl_error("d2i_ASN1_OCTET_STRING failed");
 		goto fail;
 	}
-	attrp = octet->data;
-	if (o2i_ECPublicKey(&ec, &attrp, octet->length) == NULL) {
+	attrp = ASN1_STRING_get0_data(octet);
+	if (o2i_ECPublicKey(&ec, &attrp, ASN1_STRING_length(octet)) == NULL) {
 		ossl_error("o2i_ECPublicKey failed");
 		goto fail;
 	}
